From linux-usb-owner@vger.kernel.org  Mon Oct  6 21:11:07 2008
From: Alan Stern <stern@rowland.harvard.edu>
Date: Mon, 6 Oct 2008 11:24:26 -0400 (EDT)
Subject: USB: snoop processes opening usbfs device files
To: Greg KH <greg@kroah.com>
Message-ID: <Pine.LNX.4.44L0.0810061123070.2180-100000@iolanthe.rowland.org>


This patch (as1148) adds a new "snoop" message to usbfs when a device
file is opened, identifying the process responsible.  This comes in
extremely handy when trying to determine which program is doing some
unwanted USB access.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/devio.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/usb/core/devio.c
+++ b/drivers/usb/core/devio.c
@@ -624,6 +624,8 @@ static int usbdev_open(struct inode *ino
 	smp_wmb();
 	list_add_tail(&ps->list, &dev->filelist);
 	file->private_data = ps;
+	snoop(&dev->dev, "opened by process %d: %s\n", task_pid_nr(current),
+			current->comm);
  out:
 	if (ret) {
 		kfree(ps);
